src/main/java/com/app/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+import com.app.bookstore.dto.order.OrderItemDto;
+import com.app.bookstore.dto.order.PlaceOrderRequestDto;
+import com.app.bookstore.dto.order.UpdateOrderStatusRequestDto;
+import com.app.bookstore.exceptions.EntityNotFoundException;
+import com.app.bookstore.mapper.OrderItemMapper;
+import com.app.bookstore.mapper.OrderMapper;
+import com.app.bookstore.model.CartItem;
+import com.app.bookstore.model.Order;
+import com.app.bookstore.model.OrderItem;
+import com.app.bookstore.model.ShoppingCart;
+import com.app.bookstore.model.User;
+import com.app.bookstore.repository.cart.CartItemRepository;
+import com.app.bookstore.repository.cart.ShoppingCartRepository;
+import com.app.bookstore.repository.order.OrderItemRepository;
+import com.app.bookstore.repository.order.OrderRepository;
+import com.app.bookstore.service.OrderService;
+import com.app.bookstore.service.ShoppingCartService;
+import com.app.bookstore.service.UserService;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository cartRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final UserService userService;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public OrderDto createOrder(PlaceOrderRequestDto request) {
+        ShoppingCart cart = getShoppingCart();
+        Set<CartItem> cartItems = new HashSet<>(
+                cartItemRepository.findCartItemsByShoppingCartId(cart.getId()));
+        cart.setCartItems(cartItems);
+        Order order = orderMapper.toOrder(cart, request);
+        order.setShippingAddress(request.getShippingAddress());
+        order.setTotal(calculateTotal(order.getOrderItems()));
+        Order savedOrder = saveOrderWithItems(cartItems, order);
+        shoppingCartService.clear(cart);
+        return orderMapper.toDto(savedOrder);
+    }
+
+    @Override
+    public OrderDto updateOrderStatus(Long orderId, UpdateOrderStatusRequestDto request) {
+        Order order = orderRepository.findOrderByOrderId(orderId);
+        order.setStatus(request.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public OrderItemDto getItemById(Long orderId, Long itemId) {
+        User user = userService.getAuthenticatedUser();
+        Order order = orderRepository.findOrderByOrderId(orderId);
+        if (Objects.equals(order.getUser().getId(), user.getId())) {
```

Comment:
1. Code duplication 2. It is possible to make such check on BD level, just do not fetch order which is not owned by user